tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordingHandlerTests.cs

```diff
@@ -631,6 +631,49 @@ public void TestCreateUpstreamRequestIncludesExpectedHeaders(params string[] inc
             }
         }
 
+        [Theory]
+        [InlineData("{ \"HandleRedirects\": \"true\"}", true)]
+        [InlineData("{ \"HandleRedirects\": \"false\"}", false)]
+        [InlineData("{ \"HandleRedirects\": \"1\"}", true)]
+        [InlineData("{ \"HandleRedirects\": \"0\"}", false)]
+        [InlineData("{ \"HandleRedirects\": \"True\"}", true)]
+        [InlineData("{ \"HandleRedirects\": \"False\"}", false)]
+        [InlineData("{ \"HandleRedirects\": \"TRUE\"}", true)]
+        [InlineData("{ \"HandleRedirects\": \"FALSE\"}", false)]
```

Comment:
Any reason to not make this just a JSON boolean literal, i.e. `{ "HandleRedirects": true }` and take in `Dictionary<string, object>` instead? Then you wouldn't have to parse all this stuff to see whether it's true or false.

Reply:
The intent is to use `SetRecordingOptions` for more than just HandleRedirects. Hence the additional parsing. 

I could make it string, object, and cast, but it would end up having some parsing in that situation as well. 

I could be totally missing your intent though!

Reply:
Yeah, I was just thinking that having to pass a string value (`"true"`) in instead of an actual boolean literal (`true`) to the setting feels a little strange. In practice though it's pretty minor.

Reply:
I see. You're right. I'll make it object and still handle the casting.

The goal is to make it as easy as possible for you folks.